store/src/store.rs

```diff
@@ -236,7 +250,19 @@ impl<DB: Database> KVStore<DB> {
     }
 }
 
-pub(crate) enum AnyKVStore<'a, DB: Database> {
+/// Equivalent to [`BasicKVStore`](https://docs.cosmos.network/v0.46/core/store.html#base-layer-kvstores) from cosmos
+pub trait KVStoreTrait {
+    fn get(&self, k: &impl AsRef<[u8]>) -> Option<Vec<u8>>;
+    // fn contains( &self, k : &impl AsRef<[u8]>) -> bool;
+    fn set(&mut self, key: impl IntoIterator<Item = u8>, value: impl IntoIterator<Item = u8>);
```

Comment:
We can't have a set method since that will never work on a `QueryKVStore`
We should have a range method.

Reply:
Before I add range method, we should merge [this](https://github.com/rumos-io/gears/pull/33) PR as it contains changes to range method